### ISSUE TITLE
Setting `op_type` in event metadata is not allowed anymore after migration to data streams

### ISSUE THREAD

**Issue by rdner** [elastic/beats#30085]
While trying this configuration from another issue https://github.com/elastic/beats/issues/25425:

(it's a workaround for another reported bug but the point is that it sets `op_type: "index"` for an event metadata on ingestion to Elasticsearch)

```yaml
processors:
  - fingerprint:
      fields: ["id"]
      target_field: "@metadata._id"
  - add_fields:
      target: "_tmp"
      fields:
        op_type: "index"
  - convert:
      fields:
      - from: "_tmp.op_type"
        to: "@metadata.op_type"
        type: "string"
  - drop_fields.fields:
    - "_tmp"
```

I realised that using `op_type: "index"` is not permitted anymore due to the recent migration to data streams in 8.0 (https://github.com/elastic/beats/pull/28450) and it causes this error in the filebeat output:

TLDR; 
```
{\"type\":\"illegal_argument_exception\",\"reason\":\"only write ops with an op_type of create are allowed in data streams\"}, dropping event!"
```

```json
{
  "log.level": "warn",
  "@timestamp": "2022-01-28T10:45:58.164+0100",
  "log.logger": "elasticsearch",
  "log.origin": {
    "file.name": "elasticsearch/client.go",
    "file.line": 428
  },
  "message": "Cannot index event publisher.Event{Content:beat.Event{Timestamp:time.Date(2022, time.January, 28, 10, 45, 57, 86498000, time.Local), Meta:{\"_id\":\"39e5a0c8ff10416ea4a04e433dc8c4c99b64a3d7be5355d8bfa569f391043a11\",\"op_type\":\"index\"}, Fields:{\"agent\":{\"ephemeral_id\":\"bdc9966b-d4c4-4690-87cf-2159adb29440\",\"id\":\"4fd6b173-0f13-4792-888a-b54dcd85496c\",\"name\":\"MacBook-Pro.localdomain\",\"type\":\"filebeat\",\"version\":\"8.1.0\"},\"ecs\":{\"version\":\"8.0.0\"},\"host\":{\"name\":\"MacBook-Pro.localdomain\"},\"input\":{\"type\":\"log\"},\"log\":{\"file\":{\"path\":\"/Users/rdner/Projects/tests/some.log\"},\"offset\":15},\"message\":\"ge\"}, Private:file.State{Id:\"native::14147072-16777234\", PrevId:\"\", Finished:false, Fileinfo:(*os.fileStat)(0x14000581520), Source:\"/Users/rdner/Projects/tests/some.log\", Offset:18, Timestamp:time.Date(2022, time.January, 28, 10, 45, 57, 85657000, time.Local), TTL:-1, Type:\"log\", Meta:map[string]string(nil), FileStateOS:file.StateOS{Inode:0xd7de00, Device:0x1000012}, IdentifierName:\"native\"}, TimeSeries:false}, Flags:0x1, Cache:publisher.EventCache{m:common.MapStr(nil)}} (status=400): {\"type\":\"illegal_argument_exception\",\"reason\":\"only write ops with an op_type of create are allowed in data streams\"}, dropping event!",
  "service.name": "filebeat",
  "ecs.version": "1.6.0"
}
```

As our Elasticsearch documentation states:

> If the request targets a data stream, an op_type of create is required.

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#docs-index-api-query-params

This means none of the beats in 8.0 can support setting `op_type` anymore and according to these issues below, `op_type: "index"` was an option requested by the customers (as well as `update` and `delete`) and it seems like we broke it with the migration to data streams. I assume if it was requested it is used by some of our customers.

https://github.com/elastic/beats/issues/23091
https://github.com/elastic/beats/issues/8534
https://github.com/elastic/beats/pull/12606

For confirmed bugs, please report:
- Version: 8.0
- Steps to Reproduce: 
Run a filebeat with the configuration above and try to ingest an event to Elasticsearch (an event must have an `id` field for the `fingerprint` processor to work)

**Comment by rdner** [comment]
@kvch since you worked on the migration to data streams would be nice to have your input on this.

**Comment by kvch** [comment]
It is not a bug, but a breaking change we introduced in 8.0. Data streams are designed for append-only time-series data, hence they only support `create` `op_type`. The current workaround is either to target directly the backing indices as explained here: https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html#data-streams-append-only
Alternatively, you can export the index template, remove the line containing `"data_stream"`, change the index pattern to `{beatname}-{version}*` and load it back to ES. In this case you have to set up ILM for the index using the UI in Kibana.
